src/main/java/uk/gov/hmcts/cmc/claimstore/idam/models/AuthenticateUserResponse.java

```diff
@@ -1,15 +1,19 @@
 package uk.gov.hmcts.cmc.claimstore.idam.models;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class AuthenticateUserResponse {
     @JsonProperty("default-url")
     private String defaultUser;
-    @JsonProperty("access-token")
+    private String code;
+    @JsonProperty("access_token")
     private String accessToken;
 
-    public AuthenticateUserResponse(String defaultUser, String accessToken) {
+    public AuthenticateUserResponse(String defaultUser, String code, String accessToken) {
```

Comment:
We should split this into two cases, one is `AuthorizeReposnse` and other one is `TokenExchangeResponse`